app/domain/authentication/security/process_request_proxies.rb

```diff
@@ -0,0 +1,58 @@
+require 'command_class'
+
+module Authentication
+  module Security
+    ProcessRequestProxies ||= CommandClass.new(
+      dependencies: {
+        # InputValidation
+      },
+      inputs: [:request]
+    ) do
+
+      def call
+        validate_request
+        process_xff_header
+      end
+
+      private
+
+      Struct.new('RemoteAddress', :client_ip, :proxy_list)
```

Comment:
this struct should be a [Value Object](https://revs.runtime-revolution.com/value-objects-in-ruby-on-rails-9df64bc8db34)

Reply:
how would you call the class that holds this value object?

Reply:
> this struct should be a [Value Object](https://revs.runtime-revolution.com/value-objects-in-ruby-on-rails-9df64bc8db34)
Do you mean to create a separate class in a separate file for that? 


> how would you call the class that holds this value object?
It contains info about the request so maybe RequestIpInfo?

Reply:
yes, the value object should be in a separate class.
so if it is called `RequestIpInfo` then this command class should be `ExtractRequestIpInfo` as it is more informative than `ProcessRequestHeader.